thunder/tests/test_inplace_functionalization.py

```diff
@@ -355,6 +355,33 @@ def f_with_view(a):
     torch.testing.assert_close(x, x_ref)
 
 
+@instantiate(
+    dtypes=NOTHING,
+)
+def test_multiple_views_before_inplace_to_base(executor, device, _):
+
+    # ref: https://github.com/pytorch/pytorch/blob/29e2e2a/test/test_functionalization.py#L159-L169
+    def f(x):
```

Comment:
I tried one test case following https://github.com/Lightning-AI/lightning-thunder/issues/657#issuecomment-2242979740, the main branch failed to create a correct trace but this pr seems to succeed

Reply:
The trace however doesn't seem to copy the value of `x` after `add_` to the views explicitly, so it seems like that we implicitly depend on (a) thunder preserves `prims.copy_` if an operand of in-place ops is arg.

Reply:
Hm. I'm not sure I understand this and if it is safe...